app/src/main/java/fr/free/nrw/commons/CommonsApplication.java

```diff
@@ -275,7 +275,7 @@ public void clearApplicationData(Context context, LogoutListener logoutListener)
                     //TODO: fix preference manager
                     defaultPrefs.clearAll();
                     defaultPrefs.putBoolean("firstrun", false);
-                    updateAllDatabases();
+                    new Thread(this::updateAllDatabases).start();
```

Comment:
We should not do this, a) threads are expensive and b) Rx gives us the ability to switch threads easily

You can use `andThen` to execute another completable and keep the logoutListener on the main thread.

If you have any questions about Rx then please ask me

Reply:
@macgills I never used andThen operator can you give some suggestions? Thanks!

Reply:
example:

```
Completable.fromCallable{
writeSomeValuesToADatabase()
}
.subscribeOn(Schedlers.io())
.andThen{
 Completable.fromCallable{
  writeSomeValuesToAnotherDatabase() 
}
.observeOn(AndroidSchedulers.mainThread())
.subscribe {
  updateTheUIBecauseWeAreOnTheUiThread()
}
}
```

that is kinda of kotlin-y but that is what my pseudocode tends to look like nowadays

Reply:
@macgills Should I use `Completable.fromCallable` or `Completable.differ` Thanks!

Reply:
Generally you should use `fromCallable`.

FromCallable can basically be written as

```
Completable.defer{
  Completable.just(something)
}
```

if Completable had `.just` which it doesn't because that would be a little silly because a completable doesn't represent any values but a unit of work to be done.

This is more relevant to Flowable/Observable/Single/Maybe. Essentially it allows you to supply a factory for an Observable so that whenever it gets subscribed to it creates a new one